src/index.ts

```diff
@@ -74,43 +74,62 @@ export interface AddressEmissionsResult {
   gasUsed: number;
 }
 
-/**
- * Calculate emissions of an address. Emissions are allocated for SENT (outgoing) transactions only.
- */
-export const calculateAddressEmissions = async (
+const fetchTransactions = async (
   options: CalculatorOptions
-): Promise<AddressEmissionsResult> => {
-  validateCalculatorOptions(options);
+): Promise<TransactionData[]> => {
   const response = await fetchJSON<EtherscanResponse>(
     constructEtherscanURL(options)
   );
   if (response.status === "0" && response.message === "No transactions found") {
-    return {
-      transactionType: options.transactionType,
-      kgCO2: 0,
-      transactionsCount: 0,
-      gasUsed: 0,
-    };
+    return new Array<TransactionData>();
   }
   if (response.status !== "1") {
     throw new Error(
-      `Failed to calculate address emissions: ${response.message}`
-    );
-  }
-  if (response.result.length >= 10000) {
-    throw new Error(
-      `This address has too many ${options.transactionType} transactions to count! This calculator can't handle addresses with more than 10,000 transactions of any one type.`
+      `Failed to calculate emissions: ${response.message}`
     );
+    }
+  return response.result;
+};
+
+const getTransactions = async (
+  options: CalculatorOptions
+): Promise<TransactionData[]> => {
+  const transactions = await fetchTransactions(options);
+  if (transactions.length < 10000) {
+    return transactions;
   }
-  const txns = filterValidOutgoingTransactions(
-    response.result,
-    options.address
+  const lastTransaction = transactions[transactions.length - 1];
+  const lastBlockNumber = Number(lastTransaction.blockNumber);
+  const selectedTransactions = transactions.filter(
+    transaction => Number(transaction.blockNumber) < lastBlockNumber
   );
-  const totalGasUsed = getSumGasUsed(txns);
+  const missedTransactions = await getTransactions({
+    transactionType: options.transactionType,
+    address: options.address,
+    etherscanAPIKey: options.etherscanAPIKey,
+    startBlock: lastBlockNumber,
+    endBlock: options.endBlock,
+  });
+  return selectedTransactions.concat(missedTransactions);
```

Comment:
I just learned it's slightly more performant to do:

```suggestion
  selectedTransactions.push(...missedTransactions);
  return selectedTransactions;
```

Because it mutates the same array instead of creating a new one every time